### PR TITLE
Add rancher_registration_token resource type

### PR DESCRIPTION
**Pull request by raphink**
Refactor config.go to allow for environment clients to be created

### CODE REVIEW

**Comment by blackjid** [comment]
hey @raphink awesome! thanks!
I see you added a way to recreate the client for the environment specific resources... I was doing that in the registry branch, but I couldn't make the tests work for a while... 

I think now I can rebase the registry branch from yours, and make it work.. :)

File: README.md [hunk 1]
```diff
@@ -58,3 +58,45 @@ The following attributes are exported:
...
 * `description` - The description of the environment.
 * `orchestration` - The orchestration engine for the environment.
+
+### Registration Token
+
+Provides a Rancher Registration Token resource. This can be used to create registration tokens for rancher environments and retrieve their information.
+
+#### Example Usage
+
+```
```
Comment (blackjid) [inline README.md]: what about using the hcl highlighting now you proposed that on the rest of the docs ?

File: README.md [hunk 2]
```diff
@@ -58,3 +58,45 @@ The following attributes are exported:
...
+  name = "staging_token"
+  description = "Registration token for the staging environment"
+  environment_id = "${rancher_environment.default.id}"
+}
+
+output "registration_url" {
+  value = "${rancher_registration_token.registration_url}"
+}
+
+output "registration_token" {
```
Comment (blackjid) [inline README.md]: do you think is necessary to add the outputs to the example??
If we do that will end up repeating all the attributes in the attributes references here in the example...

File: rancher/resource_rancher_registration_token.go [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+	d.Set("description", regT.Description)
+	d.Set("name", regT.Name)
+	d.Set("token", regT.Token)
+	d.Set("registration_url", regT.RegistrationUrl)
+
+	return nil
+}
+
+func resourceRancherRegistrationTokenDelete(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("[INFO] Deleting RegistrationToken is not possible, simply removing its ID: %s", d.Id())
```
Comment (blackjid) [inline rancher/resource_rancher_registration_token.go]: what do you mean with this line of log?
At the end you are effectively deleting the token right?? after deactivating it...

**Comment by raphink** [comment]
@blackjid Ah, hadn't seen your registry branch, sorry! Reorganizing this client thing to get Environment-specific clients was a pain, but required anyway for most resource types (registry, registration_token, stack, etc.)

I just rebased the PR now that you merged the other one.

File: README.md (hunk 1 above)
Comment (blackjid) [reply README.md]: we'll this is done now :)

**Comment by blackjid** [comment]
don't worry I had the branch, but some stuff were just in my computer, i've just push them...

File: README.md (hunk 1 above)
Comment (raphink) [reply README.md]: Yes, I was waiting for you to merge the other PR first.

File: README.md (hunk 2 above)
Comment (raphink) [reply README.md]: OK, I can remove those.

File: rancher/resource_rancher_registration_token.go (hunk 3 above)
Comment (raphink) [reply rancher/resource_rancher_registration_token.go]: My bad. I couldn't get deleting to work at first, so I had put this. I've now found how to deactivate/remove, so I can remove this.

**Comment by raphink** [comment]
@blackjid there you go, I fixed all your comments.

**Pull request merged**